http/request.go

```diff
@@ -167,29 +180,74 @@ func (r *Request) createHTTPRequest() (*http.Request, error) {
 }
 
 // sendRequest sends the request using the given HTTP client
-func (r *Request) sendRequest(client *http.Client, logger logger.Logger, retriesRemaining uint) (*Response, error) {
-	request, err := r.createHTTPRequest()
-	if err != nil {
+func (r *Request) sendRequest(ctx context.Context, client *http.Client, logger logger.Logger, retriesRemaining uint) (*Response, error) {
+	ctx, span := tracer.Start(ctx, r.url.Hostname()) // TODO: Decide on how to name this span
```

Comment:
I think the tracing should be off by default, only if a tracer and trace name / existing span is specified when creating the client should we trace, 
- We also need to propagate any parent span to upstream calls
- We should also have fine grained options on what to add to trace (req/response headers (sanitized of sensitive headers), request, response body (with limits for the content)

Reply:
@moshloop How would sanitizing sensitive fields from the HTTP header work? Do we ask the user for the list of HTTP headers that needs to be sanitized like this - https://www.elastic.co/guide/en/apm/agent/go/current/configuration.html#config-sanitize-field-names